packages/pubsub/src/index.js

```diff
@@ -78,27 +69,139 @@ function PubSub(options) {
     return new PubSub(options);
   }
 
-  this.defaultBaseUrl_ = 'pubsub.googleapis.com';
-  this.determineBaseUrl_(options.apiEndpoint);
-
-  var config = {
-    baseUrl: this.baseUrl_,
-    customEndpoint: this.customEndpoint_,
-    service: 'pubsub',
-    apiVersion: 'v1',
-    scopes: [
-      'https://www.googleapis.com/auth/pubsub',
-      'https://www.googleapis.com/auth/cloud-platform'
-    ],
-    packageJson: require('../package.json')
-  };
-
-  this.options = options;
+  this.options = extend({
+    scopes: v1.ALL_SCOPES
+  }, options);
 
-  commonGrpc.Service.call(this, config, options);
+  this.api = {};
+  this.auth = googleAuth(this.options);
+  this.projectId = this.options.projectId || '{{projectId}}';
```

Comment:
Yeah, `{{projectId}}` is a placeholder that is looked at as a last step before any API requests are sent. It is replaced with a given or detected value, otherwise an error is returned.